### ISSUE TITLE
I am getting and issue with JPA

### ISSUE THREAD

**Issue by brunollopes** [Saka7/spring-boot-angular4-boilerplate#5]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#63da207f': Cannot resolve reference to bean 'entityManagerFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'entityManagerFactory' available
	at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:378) ~[spring-beans-5.0.0.RC4.jar:5.0.0.RC4]
	at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveValueIfNecessary(BeanDefinitionValueResolver.java:110) ~[spring-beans-5.0.0.RC4.jar:5.0.0.RC4]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveConstructorArguments(ConstructorResolver.java:640) ~[spring-beans-5.0.0.RC4.jar:5.0.0.RC4]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:461) ~[spring-beans-5.0.0.RC4.jar:5.0.0.RC4]
	at

Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'entityManagerFactory' available

Should I need to change the backend and declare explicity the entityManagerFactory ?

Thanks

**Comment by brunollopes** [comment]
in this one I am using spring-boot 2.0.0.M4

**Comment by Saka7** [comment]
Right suggestion. Error is thrown probably because of `boot` version `2.0.0.M4` mainly `spring-beans` version `5.0.0.RC4`. Everything works with `spring-boot 1.5.2.RELEASE`.

**Comment by brunollopes** [comment]
I have changed to the 1.5.2.RELEASE and still got the same:

Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jwtUserDetailsServiceImpl': Unsatisfied dependency expressed through method 'setUserRepository' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'userRepository': Cannot create inner bean '(inner bean)#770f146b' of type [org.springframework.orm.jpa.SharedEntityManagerCreator] while setting bean property 'entityManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#770f146b': Cannot resolve reference to bean 'entityManagerFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'entityManagerFactory' available

Maybe is some missing on my side, do you have any quick tip on it ?

Thanks

**Comment by Saka7** [comment]
Can you paste your `build.gradle` file?

**Comment by Saka7** [comment]
Did you run `scripts/init_db` script? Check if `users` table has been created in your database.

**Comment by brunollopes** [comment]
Hi! I have it worked now! Thanks, but now I can't access to it.
Accessing to :http://localhost:8080/ it shows the text 'Loading...' but then only a white page is displayed.
Trying to access to the points: http://localhost:8080/api/auth/signup, returns:
Whitelabel Error Page

This application has no explicit mapping for /error, so you are seeing this as a fallback.

Sat Sep 23 18:44:14 WEST 2017
There was an unexpected error (type=Not Found, status=404).
No message available

I have checked the JAR and the client statics files are there... any ideas ?

Thanks

**Comment by Saka7** [comment]
Hi.

The page should be blank. Following template doesn't have any UI-related stuff #4. Angular part only contains auth service. 

You cannot access `/api/auth/signup` route probably because you are using browser's default `GET` method, but route is mapped to `POST` (try [postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop) or [curl](https://curl.haxx.se) or any other rest clients to test it out).

**Comment by brunollopes** [comment]
Yes thats great! but now I have implemented the interface simple form to set the username and password and email. Then I call your authservice to login and register on the server but it gaves me this issue
`
2zone.js:2744 OPTIONS http://localhost:8080/api/auth/login net::ERR_ABORTED
invokeTask @ zone.js:1427
globalZoneAwareCallback @ zone.js:1445
login?returnUrl=%2F:1 Failed to load http://localhost:8080/api/auth/login: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:4200' is therefore not allowed access. The response had HTTP status code 403.
auth.service.ts:205 Server error
`
This line 205 is the call this.http.post(AuthService.SIGNIN_URL ...

But I have tried with the headers you have added and then with others ones, but none of the options made it work. 

My server is running on port 8080 and the client on port 4200

At the moment I amusing these headers:

`
const headers = new Headers({
                                      'Access-Control-Allow-Origin' : '*',
                                      'Access-Control-Allow-Methods': 'GET, POST, PATCH, PUT, DELETE, OPTIONS',
                                     'Access-Control-Allow-Headers': 'Origin, Content-Type, X-Auth-Token',
                                     'Content-Type': 'application/json'
                                     });

`
If Itry with Postman it works well just by selecting the option for the heards: Content-Type=application/json

But is should be something else ... any idea ?

Thanks
